File: app-miyou/src/main/java/com/miyou/app/infrastructure/credit/repository/UserCreditMongoRepository.java
package com.miyou.app.infrastructure.credit.repository;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.credit.document.UserCreditDocument;
import reactor.core.publisher.Mono;

public interface UserCreditMongoRepository
	extends
		ReactiveMongoRepository<UserCreditDocument, String> {

	Mono<UserCreditDocument> findByUserId(String userId);
}

File: backend/infrastructure/src/main/kotlin/com/miyou/app/infrastructure/credit/repository/UserCreditMongoRepository.kt
package com.miyou.app.infrastructure.credit.repository

import com.miyou.app.infrastructure.credit.document.UserCreditDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Mono

interface UserCreditMongoRepository : ReactiveMongoRepository<UserCreditDocument, String> {
    fun findByUserId(userId: String): Mono<UserCreditDocument>
}
